core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java

```diff
@@ -764,30 +776,59 @@ public final void setIdleTimeout(final int idleTimeoutMillis) {
     }
 
     /**
-     * Gets the amount of time in milliseconds after which a sender link will be closed when no message has been sent on
-     * it.
+     * Gets the amount of time after which a sender link will be closed when no message has been sent on it. The time
+     * unit is set by {@link #setInactiveLinkTimeoutUnit(TimeUnit)}.
      * <p>
-     * The default value of this property is <code>0</code>, which disables automatic link timeout.
+     * The default value of this property is {@link #DEFAULT_INACTIVE_LINK_TIMEOUT}, which disables automatic link
+     * timeout.
      *
-     * @return The inactiveLinkTimeout in milliseconds.
+     * @return The amount of time to wait for inactive links.
+     * @see #getInactiveLinkTimeoutUnit()
      */
-    public long getInactiveLinkTimeout() {
+    public final long getInactiveLinkTimeout() {
```

Comment:
FMPOV it would be nice to be able to get the timeout as a `java.time.Duration` which includes both, the amount as well as the unit and provides methods for getting the duration as minutes, hours, millis etc.

The setters would be used by Spring to set the values, but the Duration could then be used much easier in client code. WDYT?

Reply:
This seems to be supported out-of-the-box already in Spring Boot: https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config-conversion-duration

Reply:
I like the idea. Even though I prefer to use the method `getInactiveLinkTimeoutMillis()` you proposed above.